tests/phpunit/integration/Core/Authentication/Google_ProxyTest.php

```diff
@@ -146,21 +201,40 @@ public function test_exchange_site_code() {
 		);
 
 		// Stub the response to the proxy oauth API.
+		$this->expect_request_response(
+			$google_proxy->url( Google_Proxy::OAUTH2_SITE_URI ),
+			$expected_credentials
+		);
+
+		$credentials = $google_proxy->exchange_site_code( 'test-site-code', 'test-undelegated-code' );
+		$this->assertEqualSetsWithIndex(
+			$expected_credentials,
+			$credentials
+		);
+	}
+
+	/**
+	 * Adds a 'pre_http_request' filter that will ensure the request for the
+	 * given URL will yield a specific response.
+	 *
+	 * @param string $request_url   Request URL to modify response for.
+	 * @param array  $response_data Response data to return for the request. Will be JSON-encoded.
+	 * @param int    $response_code Optional. Response status code to return. Default 200.
+	 */
+	private function expect_request_response( $request_url, array $response_data, $response_code = 200 ) {
```

Comment:
The name of this method is a little confusing as it reads a bit like an assertion when in fact it's setting up a mocked response for a request. Something like `mock_http_request` would be a bit more clear I think, but something else without `expect` would work too.

It would be nice to add something like this to the base test case at some point as well but that needs a bit more thought still.